node_modules/oae-authentication/lib/api.js

```diff
@@ -753,6 +751,41 @@ var _associateLoginId = function(loginId, userId, callback) {
 /**
  * Get the login ids that are mapped to a user.
  *
+ * @param  {Context}     ctx                 Standard context object containing the current user and the current tenant
+ * @param  {String}      userId              The id of the user
+ * @param  {Function}    callback            Standard callback function
+ * @param  {Object}      callback.err        An error that occurred, if any
+ * @param  {Object}      callback.loginIds   A hash whose keys are the authentication providers, and values are the LoginId objects that are mapped to the user
+ */
+var getUserLoginIds = module.exports.getUserLoginIds = function(ctx, userId, callback) {
+    // Parameter validation
+    var validator = new Validator();
+    validator.check(null, {'code': 401, 'msg': 'You have to be logged in to request the login ids for a user'}).isLoggedInUser(ctx);
+    validator.check(userId, {'code': 400, 'msg': 'A user id must be provided'}).isUserId();
+    if (validator.hasErrors()) {
+        return callback(validator.getFirstError());
+    }
+
+    // Request the user details
+    PrincipalsAPI.getUser(ctx, userId, function(err, user) {
+        if (err) {
+            return callback(err);
+        }
+
+        // Only global administrators and administrators of the tenant the user belongs to can request the login ids
+        var isGlobalAdmin = (ctx.user() && ctx.user().isGlobalAdmin());
+        var isAdmin = (ctx.user().isAdmin(user.tenant.alias));
+        if (!isGlobalAdmin && !isAdmin) {
+            return callback({'code': 401, 'msg': 'You are not authorized to request the login ids for this user'});
+        }
```

Comment:
A few things about this check:

1. We don't have to check if `ctx.user()` is defined, because `isLoggedInUser` will have already ensured that
2. `ctx.user().isAdmin(user.tenant.alias)` will also take into consideration `isGlobalAdmin`, so we just need the one check
3. The check should be done in an `else if` of the prior error handling if statement